src/types/Slot0.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+/**
+ * @dev Slot0 is a packed version of solidity structure.
+ * Using the packaged version saves gas by not storing the structure fields in memory slots.
+ *
+ * Layout:
+ * 24 bits empty | 24 bits lpFee | 24 bits protocolFee | 24 bits tick | 160 bits sqrtPriceX96
+ *
+ * Fields in the direction from the least significant bit:
+ *
+ * The current price
+ * uint160 sqrtPriceX96;
+ *
+ * The current tick
+ * int24 tick;
+ *
+ * Protocol fee, expressed in hundredths of a bip, upper 12 bits are for 1->0, and the lower 12 are for 0->1
+ * the maximum is 1000 - meaning the maximum protocol fee is 0.1%
+ * the protocolFee is taken from the input first, then the lpFee is taken from the remaining input
+ * uint24 protocolFee;
+ *
+ * Used for the lp fee, either static at initialize or dynamic via hook
+ * uint24 lpFee;
+ */
+type Slot0 is bytes32;
+
+using Slot0Library for Slot0 global;
+
+library Slot0Library {
+    uint160 internal constant UINT160_MASK = 0x00FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF;
+    uint24 internal constant UINT24_MASK = 0xFFFFFF;
```

Comment:
nit: would rather this was called `24_BIT_MASK` or something like that now that its used for both int24 and uint24

Reply:
Changed to `MASK_24_BITS`